here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/WriteFeatureApiTask.java

```diff
@@ -228,6 +233,198 @@ protected void init() {}
+      }
+      try {
+        featuresToPatchFromStorage = readFeaturesFromResult(result, XyzFeature.class, 0, DEF_FEATURE_LIMIT);
+      } catch (NoCursor | NoSuchElementException emptyException) {
+        if (responseType.equals(HttpResponseType.FEATURE)) {
+          // If this is patching only 1 feature (PATCH by ID), return not found
+          return returnError(
+              "Unexpected null result while reading current versions in storage of targeted features for PATCH. The feature does not exist.",
+              "Feature does not exist.",
+              XyzError.NOT_FOUND);
+        } else if (!responseType.equals(HttpResponseType.FEATURE_COLLECTION)) {
+          // This function was then misused somewhere. FIND AND FIX IT!!
+          return returnError(
+              "Unsupported HttpResponseType was called: {}",
+              responseType,
+              "Internal server error.",
+              XyzError.EXCEPTION);
+        }
+        // Else none of the features exists in storage, will create them later
+      }
+      // As applicable, modify features based on parameters supplied
+      final QueryParameterList queryParams = queryParamsFromRequest(routingContext);
+      final List<String> addTags = extractParamAsStringList(queryParams, ADD_TAGS);
+      final List<String> removeTags = extractParamAsStringList(queryParams, REMOVE_TAGS);
+      for (XyzFeature featureFromRequest : featuresFromRequest) {
+        addTagsToFeature(featureFromRequest, addTags);
+        removeTagsFromFeature(featureFromRequest, removeTags);
+      }
+      // Attempt patching, keeping the order of the features from the request
+      patchedFeature = performInMemoryPatching(featuresFromRequest, featuresToPatchFromStorage);
+    }
+    final WriteXyzFeatures wrRequest = RequestHelper.upsertFeaturesRequest(spaceId, patchedFeature);
+    // Forward request to NH Space Storage writer instance
+    try (final IWriteSession writer = naksha().getSpaceStorage().newWriteSession(context(), true)) {
+      final Result wrResult = writer.execute(wrRequest);
+      if (wrResult == null) {
+        // unexpected null response
+        writer.rollback(true);
+        writer.close();
+        return returnError(
+            "Received null result after writing patched features, rolled back.",
+            "Unexpected null result.",
+            XyzError.EXCEPTION);
+      } else if (wrResult instanceof ErrorResult er) {
+        writer.rollback(true);
+        writer.close();
+        try (ForwardCursor<XyzFeature, XyzFeatureCodec> resultCursor = er.getXyzFeatureCursor()) {
+          if (!resultCursor.hasNext()) {
+            throw new NoSuchElementException("Error Result Cursor is empty");
+          }
+          while (resultCursor.hasNext()) {
+            if (!resultCursor.next()) {
+              throw new RuntimeException("Unexpected invalid error result");
+            }
+            if (!er.reason.equals(XyzError.CONFLICT)) {
```

Comment:
great catch, thanks, my mistake using the error Result directly here. I pushed a fix now targeting correctly the current error from the Cursor. 

It sounds like the first condition is always true, so I opt for checking only the second condition, which is already in the code (albeit with that mistake of looking at the wrong error object previously)

Reply:
> It sounds like the first condition is always true, so I opt for checking only the second conditio

Going by design, this is not true, because result can have multiple features and not all of them might have failed. Right thing is to check if there was an ERROR for that feature, then only check if there was CONFLICT. Also, another suggestion is to check other way round handling the null condition ... meaning `CONFLICT.equals(err)` , knowing that err can be null as well.

Reply:
got it, I pushed the fix